src/vmspawn/vmspawn.c

```diff
@@ -1141,6 +1164,34 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
         if (r < 0)
                 return log_oom();
 
+        FOREACH_ARRAY(mount, arg_runtime_mounts.mounts, arg_runtime_mounts.n_mounts) {
+                _cleanup_free_ char *sock_path = NULL, *sock_name = NULL;
+                r = start_virtiofsd(bus, trans_scope, our_runtime_dir, mount->source, /* uidmap= */ false, &sock_path, &sock_name);
+                if (r < 0)
+                        return r;
+
+                r = strv_extend(&cmdline, "-chardev");
+                if (r < 0)
+                        return log_oom();
+
+                r = strv_extendf(&cmdline, "socket,id=%1$s,path=%2$s/%1$s", sock_name, sock_path);
+                if (r < 0)
+                        return log_oom();
+
+                r = strv_extend(&cmdline, "-device");
+                if (r < 0)
+                        return log_oom();
+
+                r = strv_extendf(&cmdline, "vhost-user-fs-pci,queue-size=1024,chardev=%1$s,tag=%1$s", sock_name);
+                if (r < 0)
+                        return log_oom();
+
+                r = strv_extendf(&arg_kernel_cmdline_extra, "systemd.mount-extra=%s:%s:virtiofs:%s",
+                                 sock_name, mount->target, mount->read_only ? "ro" : "rw");
```

Comment:
hmm, i guess this needs some escaping in case of paths with spaces

Reply:
Also needed escaping from : as well, I've implemented it now using xescape on each path with `bad = "\":"`, which should hex escape : and " making the paths safe to be inside a " string from the kernel cmdline side, and free to be separated by : for us to safely parse with extract_many_words.